state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -123,136 +172,131 @@ def __init__(self):
-            self.hardware['BurnWire'] = True
-
-    def __init_sun_sensors(self):
+    def __init_sun_sensors__(self):
+        """ Initialize sun sensors one at a time """
         sun_sensors = []
 
         try:
             sun_yn = adafruit_tsl2561.TSL2561(self.i2c2, address=0x29)  # -Y
             sun_sensors.append(sun_yn)
+            self.hardware['Sun -Y'] = True
         except Exception as e:
-            print('[ERROR][Sun Sensor -Y]', e)
+            print('[ERROR][Initializing Sun Sensor -Y]', e)
 
         try:
             sun_zn = adafruit_tsl2561.TSL2561(self.i2c2, address=0x39)  # -Z
             sun_sensors.append(sun_zn)
+            self.hardware['Sun -Z'] = True
         except Exception as e:
-            print('[ERROR][Sun Sensor -Z]', e)
+            print('[ERROR][Initializing Sun Sensor -Z]', e)
 
         try:
             sun_xn = adafruit_tsl2561.TSL2561(self.i2c1, address=0x49)  # -X
             sun_sensors.append(sun_xn)
+            self.hardware['Sun -X'] = True
         except Exception as e:
-            print('[ERROR][Sun Sensor -X]', e)
+            print('[ERROR][Initializing Sun Sensor -X]', e)
 
         try:
             sun_yp = adafruit_tsl2561.TSL2561(self.i2c1, address=0x29)  # +Y
             sun_sensors.append(sun_yp)
+            self.hardware['Sun +Y'] = True
         except Exception as e:
-            print('[ERROR][Sun Sensor +Y]', e)
+            print('[ERROR][Initializing Sun Sensor +Y]', e)
 
         try:
             sun_zp = adafruit_tsl2561.TSL2561(self.i2c1, address=0x39)  # +Z
             sun_sensors.append(sun_zp)
+            self.hardware['Sun +Z'] = True
         except Exception as e:
-            print('[ERROR][Sun Sensor +Z]', e)
+            print('[ERROR][Initializing Sun Sensor +Z]', e)
 
         try:
             sun_xp = adafruit_tsl2561.TSL2561(self.i2c2, address=0x49)  # +X
             sun_sensors.append(sun_xp)
+            self.hardware['Sun +X'] = True
         except Exception as e:
-            print('[ERROR][Sun Sensor +X]', e)
+            print('[ERROR][Initializing Sun Sensor +X]', e)
 
         for i in sun_sensors:
             i.enabled = False  # set enabled status to False
 
-        return sun_sensors
+        self.sun_sensors = sun_sensors
 
-    def __init_coil_drivers(self):
-        coils = []
+    def __init_coil_drivers__(self):
+        """ Initialize coil drivers one at a time """
+        coil_drivers = []
 
         try:
-            drv_x = drv8830.DRV8830(self.i2c3, 0x68)  # U6
-            coils.append(drv_x)
+            # may need to fix i2c addresses
+            # schematic says U7 -> 0xC4 and 0xC5 but these vals are 8 bit instead of 7
+            drv_x = drv8830.DRV8830(self.i2c1, 0x68)  # U7
+            coil_drivers.append(drv_x)
+            self.hardware['Coil X'] = True
         except Exception as e:
-            print('[ERROR][H-Bridge U6]', e)
+            print('[ERROR][Initializing H-Bridge U6]', e)
```

Comment:
Should this be U7?

Reply:
It should! I must've fixed it in the comments and forgotten to change the print statements